src/solvers/smt/smt_conv.cpp

```diff
@@ -239,13 +239,59 @@ smt_astt smt_convt::convert_ast(const expr2tc &expr)
   smt_cachet::const_iterator cache_result = smt_cache.find(expr);
   if(cache_result != smt_cache.end())
     return (cache_result->ast);
+
+  /* Vectors!
+   *
+   * Here we need special attention for Vectors, because of the way
+   * they are encoded, an vector expression can reach here with binary
+   * operations that weren't done.
+   *
+   * The simplification module take care of all the operations, but if
+   * for some reason we would like to run ESBMC without simplifications
+   * then we need to apply it here.
+  */
+  if(is_vector_type(expr))
+  {
+    if(is_neg2t(expr))
+    {
+      return convert_ast(vector_type2t::distribute_operation(
+        expr->expr_id, to_neg2t(expr).value));
+    }
+    if(is_bitnot2t(expr))
+    {
+      return convert_ast(vector_type2t::distribute_operation(
+        expr->expr_id, to_bitnot2t(expr).value));
+    }
+
+    std::shared_ptr<ieee_arith_2ops> ops;
+    ops = std::dynamic_pointer_cast<ieee_arith_2ops>(expr);
```

Comment:
You can get the operands by calling expr->get_sub_expr(0) (op1), expr->get_sub_expr(1) (op2), etc.

Reply:
Is there a way to get the rounding mode here?